### PR TITLE
Added the endpoint to return the list of requests to join a given league

### PR DESCRIPTION
**Pull request by pjgaudr**

### CODE REVIEW

File: src/main/java/org/nnhl/resources/LeagueResource.java [hunk 1]
```diff
@@ -148,6 +149,26 @@ public Response getLeague(
...
 
+    @GET
+    @Path("{leagueId}/requests")
+    @ApiOperation(value = "Return the list of requests to join the specified league", authorizations = @Authorization(value = "jwt-auth"))
+    @ApiResponses(value =
+    { @ApiResponse(code = 200, message = "Requests returned successfully.", response = Request.class),
+            @ApiResponse(code = 403, message = "User is not authorized."),
+            @ApiResponse(code = 404, message = "League does not exist.") })
+    @Timed
+    public Response getLeagueRequests(
```
Comment (ladoe00) [inline src/main/java/org/nnhl/resources/LeagueResource.java]: We should protect this resource with ADMIN or MANAGER role....  I need to improve the handling of MANAGER role for  a specific league.

File: src/main/java/org/nnhl/resources/LeagueResource.java [hunk 2]
```diff
@@ -204,7 +225,11 @@ public Response joinLeague(
                 return Responses.notFound("Player does not exist");
             }
             System.out.println(principal.getEmail());
+            
             leagueDao.joinLeague(player, league, subscription);
+            // Now that the player has joined the league, delete the request to join that league
+            leagueDao.deleteLeagueRequest(leagueId, playerId);
```
Comment (ladoe00) [inline src/main/java/org/nnhl/resources/LeagueResource.java]: Let's make this part of a transaction.  If the delete fails, the join should not be committed.  I have an example in one of my DAO.

**Review by ladoe00 — CHANGES_REQUESTED** [review]

**Comment by pjgaudr** [comment]
Added @Transaction and @RolesAllowed annotations!

**Comment by pjgaudr** [comment]
I can't make calls using swagger anymore for protected resources???

**Pull request merged**